### PR TITLE
SM112+SM128 External view conversation 

### PR DESCRIPTION
**Pull request by AdamHawtin**
Acceptance tests for features added in this PR for ras-frontstage:
[ras-frontstage: SM112+SM128 External view conversation #294](https://github.com/ONSdigital/ras-frontstage/pull/294)
- View a list a conversations in the messages tab.
- View a conversation by thread_id.
- Reply to a message in the conversation view.

**How to Review:**
Run the linked branch of frontstage locally and other services on master (local or docker dev). 

Trello cards:
[SM128](https://trello.com/c/D3TwouTh/146-sm128-ext-view-full-conversation-13)
[SM112](https://trello.com/c/YKJU9WpQ/144-sm112-external-user-to-view-a-list-of-conversations-13)

Confluence pages:
[SM128](https://digitaleq.atlassian.net/wiki/spaces/RASB/pages/225705986/SM128+External+User+Views+and+replies+to+a+Conversation)
[SM112](https://digitaleq.atlassian.net/wiki/spaces/RASB/pages/224460888/SM112%3A+External+View+of+a+list+of+Conversations)

### CODE REVIEW

File: acceptance_tests/features/steps/external_conversation.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+@then('they are able to preview the first 80 characters (respecting word boundaries) of the latest message in the conversation')  # NOQA
+def preview_summary_of_conversation(_):
+    assert external_conversation.get_summary_length() < 80
+    assert external_conversation.get_endswith_ellipsis()[-3:] == '...'
+
+
+@then('the user will be able to view the conversation subject and the date and time the latest message was received')
+def view_date_and_time_of_conversation(_):
+    assert external_conversation.get_datetime() == 'Today'
```
Comment (gavinleeedwards) [inline acceptance_tests/features/steps/external_conversation.py]: How could this ever fail?  external_conversation.get_datetime() always returns string 'Today'.  Also, it doesn't test that the user can see the subject.

File: acceptance_tests/features/pages/external_conversation.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+def get_datetime():
+    return 'Today'
+
+
+def get_summary_length():
+    return len(browser.find_by_id('message-summary-1'))
+
+
+def get_endswith_ellipsis():
```
Comment (insacuri) [inline acceptance_tests/features/pages/external_conversation.py]: This should be either the test 
`return external_conversation.get_endswith_ellipsis()[-3:] == '...'` returning True/False. 
Or this should be renamed to something like 'get_subject_text'

File: acceptance_tests/features/view_and_reply_conversation_external.feature [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  Scenario: User is able to enter text up to and including 10,000 characters in the body of the message
+    Given the external user has a conversation
+    When they reply in that conversation
+    Then they are able able to enter up to and including 10,000 characters in the body of their reply
+
+  @sm128_s08
+  Scenario: User is only able to reply to a conversation if there is text in the body of their reply.
+    Given the external user has a conversation
+    When they enter text into the body of their reply (external)
+    Then they are able to reply to the conversation
```
Comment (gavinleeedwards) [inline acceptance_tests/features/view_and_reply_conversation_external.feature]: Are the When and Then repeating exactly the same steps?

File: acceptance_tests/features/pages/external_conversation.py (hunk 2 above)
Comment (dcdarrell9) [reply acceptance_tests/features/pages/external_conversation.py]: This is for making sure that the message body gets truncated and the words at the end are cut off and we get the '...' Not quite sure what you're asking?

File: acceptance_tests/features/steps/external_conversation.py (hunk 1 above)
Comment (dcdarrell9) [reply acceptance_tests/features/steps/external_conversation.py]: My bad 👍

File: acceptance_tests/features/view_and_reply_conversation_external.feature (hunk 3 above)
Comment (AdamHawtin) [reply acceptance_tests/features/view_and_reply_conversation_external.feature]: Yes, I'll redo the steps for that scenario

File: acceptance_tests/features/pages/external_conversation.py (hunk 2 above)
Comment (insacuri) [reply acceptance_tests/features/pages/external_conversation.py]: The function name is get_endswith_ellipsis.  At a glance I would expect this to return a true/false value, as it checks some text ending with ... .  This isn't what this does, this function simply gets some text and returns it, which isn't what the function name describes.
My suggestions are either change the name to make the description of it more accurate (probably the easier option).  Or do the comparison in this function and return a true/false value to assert inside the step.
Does that make sense?

**Pull request merged**